stubs/qrcode/qrcode/main.pyi

```diff
@@ -51,70 +56,72 @@ GenericImageLocal = TypeVar("GenericImageLocal", bound=BaseImage)  # noqa: Y001
     box_size: int
     border: int
     image_factory: type[GenericImage] | None
     @overload
     def __init__(
         self,
         version: ConvertibleToInt | None,
-        error_correction: Literal[0, 1, 2, 3],
+        error_correction: ErrorCorrect,
         box_size: ConvertibleToInt,
         border: ConvertibleToInt,
         image_factory: type[GenericImage],
-        mask_pattern: _MaskPattern | None = None,
+        mask_pattern: MaskPattern | None = None,
     ) -> None: ...
     @overload
     def __init__(
         self,
         version: ConvertibleToInt | None = None,
-        error_correction: Literal[0, 1, 2, 3] = 0,
+        error_correction: ErrorCorrect = 0,
         box_size: ConvertibleToInt = 10,
         border: ConvertibleToInt = 4,
         *,
         image_factory: type[GenericImage],
-        mask_pattern: _MaskPattern | None = None,
+        mask_pattern: MaskPattern | None = None,
     ) -> None: ...
     @overload
     def __init__(
         self: QRCode[_DefaultImage],
         version: ConvertibleToInt | None = None,
-        error_correction: Literal[0, 1, 2, 3] = 0,
+        error_correction: ErrorCorrect = 0,
         box_size: ConvertibleToInt = 10,
         border: ConvertibleToInt = 4,
         image_factory: None = None,
-        mask_pattern: _MaskPattern | None = None,
+        mask_pattern: MaskPattern | None = None,
     ) -> None: ...
     @property
     def version(self) -> int: ...
     @version.setter
     def version(self, value: ConvertibleToInt | None) -> None: ...
     @property
-    def mask_pattern(self) -> _MaskPattern | None: ...
+    def mask_pattern(self) -> MaskPattern | None: ...
     @mask_pattern.setter
-    def mask_pattern(self, pattern: _MaskPattern | None) -> None: ...
+    def mask_pattern(self, pattern: MaskPattern | None) -> None: ...
     modules_count: int
-    data_cache: Incomplete
-    data_list: Incomplete
+    data_cache: list[int]
+    data_list: list[QRData]
     def clear(self) -> None: ...
     def add_data(self, data: QRData | bytes | str, optimize: int = 20) -> None: ...
     def make(self, fit: bool = True) -> None: ...
-    def makeImpl(self, test, mask_pattern) -> None: ...
-    def setup_position_probe_pattern(self, row, col) -> None: ...
-    def best_fit(self, start: Incomplete | None = None): ...
-    def best_mask_pattern(self): ...
-    def print_tty(self, out: Incomplete | None = None) -> None: ...
-    def print_ascii(self, out: Incomplete | None = None, tty: bool = False, invert: bool = False): ...
+    def makeImpl(self, test: bool, mask_pattern: MaskPattern) -> None: ...
+    def setup_position_probe_pattern(self, row: int, col: int) -> None: ...
+    def best_fit(self, start: int | None = None) -> int: ...
+    def best_mask_pattern(self) -> int: ...
+    def print_tty(self, out: TextIO | None = None) -> None: ...
+    def print_ascii(self, out: TextIO | None = None, tty: bool = False, invert: bool = False) -> None: ...
```

Comment:
This should use a simple protocol instead of `TextIO`. Something like:

```python
from typing import Protocol, type_check_only

@type_check_only
class _TTYWriter(Protocol):
    def isatty(self) -> bool: ...
    def write(self, s: str, /) -> object: ...
    def flush(self) -> object: ...
```